azure-pipelines.yml

```diff
@@ -3,19 +3,112 @@
 #
 ---
 trigger:
-- master
+  branches:
+    include:
+      - master
+  tags:
+    include:
+      - "*"
 
-pool:
-  vmImage: "ubuntu-latest"
-
-steps:
-- task: NodeTool@0
-  inputs:
-    versionSpec: "12.x"
-  displayName: Use Node.js 12.x
-- script: npm ci
-  displayName: Install dependencies
-- script: npm audit
-  displayName: Audit dependencies
-- script: npm test
-  displayName: Run unit tests
+stages:
+  - stage: unit_tests
+    displayName: Unit tests
+    jobs:
+      - job: unit_tests
+        displayName: Unit tests
+        strategy:
+          matrix:
+            Linux:
+              imageName: "ubuntu-latest"
+            macOS:
+              imageName: "macOS-latest"
+            Windows:
+              imageName: "windows-latest"
+        pool:
+          vmImage: $(imageName)
+        steps:
+          - task: NodeTool@0
+            inputs:
+              versionSpec: "10.x"
+            displayName: Use Node.js 10.x
+          - script: npm ci
+            displayName: Install dependencies
+          - script: npm audit
```

Comment:
do you want to make this more robust as it sometimes fails

Reply:
Good idea, but I think I'll tackle this in a subsequent PR.